crypto/src/keys.rs

```diff
@@ -150,11 +190,34 @@ impl Diversifier {
     }
 
     #[allow(non_snake_case)]
-    pub fn derive_transmission_key(&self, ivk: IncomingViewingKey) -> TransmissionKey {
+    pub fn derive_transmission_key(&self, ivk: &IncomingViewingKey) -> TransmissionKey {
         let B_d = self.diversified_generator();
         TransmissionKey(ivk.0 * B_d)
     }
 }
 
 /// Represents a (diversified) transmission key.
+#[derive(Copy, Clone)]
 pub struct TransmissionKey(pub decaf377::Element);
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use rand_core::OsRng;
+
+    use crate::addresses::PaymentAddress;
+
+    #[test]
+    fn complete_key_generation_happy_path() {
+        let mut rng = OsRng;
+        let diversifier = Diversifier::generate(&mut rng);
+        let sk = SpendingKey::generate(&mut rng);
+        let expanded_sk = ExpandedSpendingKey::derive(&sk);
```

Comment:
general comment: I wonder if we want to have separate types for the spending key and expanded spending key, or whether we could use an approach like the one I implemented for `decaf377-rdsa`, where the signing key type contains both the secret key as well as a cached copy of the public key: https://github.com/penumbra-zone/decaf377-rdsa/blob/main/src/signing_key.rs#L18-L21

Reply:
I like that idea, especially given how closely related the two keys are - I didn't make the change in this PR but filed #76 for this so we don't lose this feedback